components/create/editor/ComboEditor.tsx

```diff
@@ -0,0 +1,656 @@
+    const sanitizedValue = value.replace(/[^0-9.]/g, '').replace(/(\..*?)\./g, '$1');
+    
+    // If completely empty, default to "0"
+    if (sanitizedValue === "") {
+      setForm(prev => ({ ...prev, liquidityAmount: "0" }));
+      return;
+    }
+    
+    setForm(prev => ({ ...prev, liquidityAmount: sanitizedValue }));
+  };
+
+  // Handle key press to prevent invalid characters
+  const handleKeyDown = (e: React.KeyboardEvent<HTMLInputElement>) => {
+    // Allow control keys and navigation
+    if (['Backspace', 'Delete', 'Tab', 'Escape', 'Enter', 'Home', 'End', 'ArrowLeft', 'ArrowRight', 'ArrowUp', 'ArrowDown'].includes(e.key) ||
+        // Allow Ctrl+A, Ctrl+C, Ctrl+V, Ctrl+X
+        (e.ctrlKey && ['a', 'c', 'v', 'x'].includes(e.key))) {
+      return;
+    }
+    // Only allow numbers and decimal point
+    if (!/[0-9.]/.test(e.key)) {
+      e.preventDefault();
+    }
+  };
+
+  // Deploy combinatorial pool
+  const { isLoading: isTransactionLoading, send: deployPool, fee } = useExtrinsic(
+    () => {
+      if (!isRpcSdk(sdk) || form.selectedMarkets.length !== 2) {
+        return;
+      }
+
+      const marketIds = form.selectedMarkets.map(m => m.marketId);
+      const liquidityNum = parseFloat(form.liquidityAmount || "0");
+      const safeLiquidity = isNaN(liquidityNum) ? 0 : Math.max(0, liquidityNum);
+      const liquidityAmount = new Decimal(safeLiquidity).mul(10 ** 10).toFixed(0);
+      const spotPricesFormatted = form.spotPrices.map(price => {
+        const numPrice = parseFloat(price || "0");
+        const safePrice = isNaN(numPrice) ? 0 : Math.max(0, numPrice); // Ensure non-negative
+        return new Decimal(safePrice).div(100).mul(10 ** 10).toFixed(0);
+      });
+      const swapFeeFormatted = new Decimal(form.swapFee).div(100).mul(10 ** 10).toFixed(0);
+
+      return sdk.api.tx.neoSwaps.deployCombinatorialPool(
+        outcomeCombinations.length,
+        marketIds,
+        liquidityAmount,
+        spotPricesFormatted,
+        swapFeeFormatted,
+        { total: 16, consumeAll: true } // Default fuel
+      );
+    },
+    {
+      onSuccess: (data) => {
+        notificationStore.pushNotification("Combinatorial pool deployed successfully!", {
+          type: "Success",
+        });
+        
+        // Extract pool ID from events and redirect
+        const poolDeployedEvent = data.events.find(
+          (event) => event.event.section === "neoSwaps" && event.event.method === "CombinatorialPoolDeployed"
+        );
+        
+        if (poolDeployedEvent) {
+          const eventData = poolDeployedEvent.event.data;
+          console.log("Pool deployed event data:", eventData);
+          
+          if (!eventData || eventData.length < 3) {
+            console.error("Invalid pool deployment event data structure");
+            return;
+          }
+          
+          const poolId = eventData[2].toString();
+          router.push(`/combo/${poolId}`);
```

Comment:
Interesting, so we are not using the indexer currently to get the pool id from a combinatorial token.